Riot/Modules/MatrixKit/Utils/MXKTools.m

```diff
@@ -69,8 +69,9 @@ + (void)initialize
         httpLinksRegex = [NSRegularExpression regularExpressionWithPattern:@"(?i)\\b(https?://\\S*)\\b" options:NSRegularExpressionCaseInsensitive error:nil];
         htmlTagsRegex  = [NSRegularExpression regularExpressionWithPattern:@"<(\\w+)[^>]*>" options:NSRegularExpressionCaseInsensitive error:nil];
         linkDetector = [NSDataDetector dataDetectorWithTypes:NSTextCheckingTypeLink error:nil];
-                
-        NSString *permalinkPattern = [NSString stringWithFormat:@"%@%@", BuildSettings.clientPermalinkBaseUrl ?: kMXMatrixDotToUrl, kMXKToolsRegexStringForPermalink];
+               
+        // if we have a custom clientPermalinkBaseUrl, we also need to support matrix.to permalinks
+        NSString *permalinkPattern = [NSString stringWithFormat:@"(?:%@|%@)%@", BuildSettings.clientPermalinkBaseUrl, kMXMatrixDotToUrl, kMXKToolsRegexStringForPermalink];
```

Comment:
@nimau why do you define a Non-capturing group here (?:...)?
I would expect "(%@|%@)%@"

Reply:
@giomfo, we don't need to capture it. We just want to make sure it only matches matrix.to or the custom base url.